src/main/scala/org/apache/daffodil/DaffodilSaver.scala

```diff
@@ -40,8 +40,16 @@ object DaffodilSaver {
    */
   def main(args: Array[String]): Unit = {
 
-    if (args.length != 4) System.err.println(s"[error] four arguments are required")
+    if (args.length != 4) {
+      System.err.println(s"four arguments are required")
```

Comment:
How does a user find out what the 4 arguments are, assuming they forgot one? This is an sbt plugin, so ... should this suggest see the README file, or other documentation? Or can it include the information in the message, perhaps in the compact form of just giving logical names to the 4 args that are needed?

Reply:
Users should never call this directly. DaffodilSaver.scala should always be forked by DaffodilPlugin.scala. So the only way for the length to not be correct is if we update DaffodilPlugin.scala to fork the saver with a different number of arguments but don't update DaffodilSaver.scala to match. Really this wants to be something like an assert.invariant, but I don't think SBT has a good way to do that.

Reply:
I would suggest just define an Assert object and add an invariant or invariantFailed method that does what this code does now. That makes it self documenting what is going on.

Reply:
Turns out scala has built-in assert functions: https://scala-lang.org/api/2.12.19/scala/Predef$.html. I've updated to use that.